### PR TITLE
add `margin` option

### LINKED ISSUE #1: Margin option
It might be useful with a `margin` option. I don't personally need it, but would welcome a pull request.

### PR DESCRIPTION
**Pull request by beatfreaker**
Fixes #1.

please review it and tell is this the optimal solution or should be implemented in any other way.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -52,7 +63,12 @@ Color of the box border.
...
 
-Space between the text and box border.
+##### margin
+
+Type: `number`, `object`  
+Default: `0`
+
+Space around the box border.
 
 Accepts a number or an object with any of the `top`, `right`, `bottom`, `left` properties. When a number is specified, the left/right padding is 3 times the top/bottom to make it look nice.
```
Comment (sindresorhus) [inline readme.md]: This needs to be preserved in the `padding` docs too.

File: test.js [hunk 2]
```diff
@@ -41,3 +41,19 @@ test('padding option - advanced', t => {
 └──────────────────┘
 	`);
 });
+
+test('padding option', t => {
```
Comment (sindresorhus) [inline test.js]: `padding` => `margin`

File: test.js [hunk 3]
```diff
@@ -41,3 +41,19 @@ test('padding option - advanced', t => {
...
+      ┌───────────────┐      
+      │               │      
+      │               │      
+      │      foo      │      
+      │               │      
+      │               │      
+      └───────────────┘      
+      
+      
+
```
Comment (sindresorhus) [inline test.js]: Too many `\n` on the bottom margin.

File: index.js [hunk 4]
```diff
@@ -33,6 +34,22 @@ module.exports = function (text, opts) {
...
+			left: margin * 3
+		};
+	} else {
+		margin = objectAssign({
+			top: 0,
+			right: 0,
+			bottom: 0,
+			left: 0
+		}, margin);
+	}
```
Comment (sindresorhus) [inline index.js]: Wonder if we could consolidate the `passing` & `margin` logic here somehow. Too much duplication.

File: index.js [hunk 5]
```diff
@@ -49,15 +66,15 @@ module.exports = function (text, opts) {
...
+	var top = colorizeBorder(repeating('\n', margin.top) + repeating(' ', margin.left) + '┌' + horizontal + '┐' + repeating(' ', margin.right));
+	var bottom = colorizeBorder(repeating(' ', margin.left) + '└' + horizontal + '┘' + repeating('\n', margin.bottom) + repeating(' ', margin.right));
 	var side = colorizeBorder('│');
 
 	var middle = lines.map(function (line) {
 		var paddingLeft = repeating(' ', padding.left);
 		var paddingRight = repeating(' ', contentWidth - stringWidth(line) - padding.left);
 
-		return side + paddingLeft + line + paddingRight + side;
+		return repeating(' ', margin.left) + side + paddingLeft + line + paddingRight + side + repeating(' ', margin.right);
```
Comment (sindresorhus) [inline index.js]: I would drop adding the actual right margin as it just makes it break lines earlier on narrow terminal windows. It has no effect anyways. Should still be handled in the options handling, but not added to the output.

File: index.js (hunk 4 above)
Comment (beatfreaker) [inline index.js]: how about creating a function which returns the object? something like:
```js
var getObject = function (detail){
var obj;
if (typeof detail=== 'number') {
		obj = {
			top: detail,
			right: detail* 3,
			bottom: detail,
			left: detail* 3
		};
	} else {
		obj = objectAssign({
			top: 0,
			right: 0,
			bottom: 0,
			left: 0
		}, detail);
	}
    return obj;
};
```

Comment (sindresorhus) [inline index.js]: :+1:

**Comment by beatfreaker** [comment]
@sindresorhus please check.

**Comment by sindresorhus** [comment]
Looks good. Thank you :)

**Pull request closed without merging**